frappe/__init__.py

```diff
@@ -713,6 +713,23 @@ def sendmail(
 allowed_http_methods_for_whitelisted_func = {}
 
 
+def apply_validate_argument_types_wrapper(func):
```

Comment:
It would be nice if we can use this on any python function as a decorator with a shorter name

Reply:
Eh just call it `validate_types`?

Don't think there's any collision or confusion with this short name  ^.

Reply:
`validate_argument_types` will be more descriptive. Will go ahead with that.